website/source/docs/guides/monitoring-telegraf.html.md

```diff
@@ -0,0 +1,257 @@
+
+```ini
+[[inputs.cpu]]
+  percpu = true
+  totalcpu = true
+  collect_cpu_time = false
+
+[[inputs.disk]]
+  # mount_points = ["/"]
+  # ignore_fs = ["tmpfs", "devtmpfs"]
+
+[[inputs.diskio]]
+  # devices = ["sda", "sdb"]
+  # skip_serial_number = false
+
+[[inputs.kernel]]
+  # no configuration
+
+[[inputs.linux_sysctl_fs]]
+  # no configuration
+
+[[inputs.mem]]
+  # no configuration
+
+[[inputs.net]]
+  interfaces = ["enp0s*"]
+
+[[inputs.netstat]]
+  # no configuration
+
+[[inputs.processes]]
+  # no configuration
+
+[[inputs.swap]]
+  # no configuration
+
+[[inputs.system]]
+  # no configuration
+```
+
+Another useful plugin is the [procstat][telegraf-procstat-input] plugin, which reports metrics for processes you select:
+
+```ini
+[[inputs.procstat]]
+  pattern = "(consul)"
+```
+
+Telegraf even includes a [plugin][telegraf-consul-input] that monitors the health checks associated with the Consul agent, using Consul API to query the data.
+
+It's important to note: the plugin itself will not report the telemetry, Consul will report those stats already using StatsD protocol.
+
+```ini
+[[inputs.consul]]
+  address = "localhost:8500"
+  scheme = "http"
+```
+
+## Telegraf Configuration for Consul
+
+Asking Consul to send telemetry to Telegraf is as simple as adding a `telemetry` section to your agent configuration:
+
+```json
+{
+  "telemetry": {
+    "dogstatsd_addr": "localhost:8125",
+    "disable_hostname": true
+  }
+}
+```
+
+As you can see, we only need to specify two options. The `dogstatsd_addr` specifies the hostname and port of the
+statsd daemon.
+
+Note that we specify DogStatsD format instead of plain statsd, which tells Consul to send [tags][tagging]
+with each metric. Tags can be used by Grafana to filter data on your dashboards (for example, displaying only
+the data for which `role=consul-server`. Telegraf is compatible with the DogStatsD format and allows us to add
+our own tags too, as you'll see later.
+
+The second option tells Consul not to insert the hostname in the names of the metrics it sends to statsd, since the hostnames will be sent as tags. Without this option, the single metric `consul.raft.apply` would become multiple metrics:
+
+        consul.server1.raft.apply
+        consul.server2.raft.apply
+        consul.server3.raft.apply
+
+If you are using a different agent (e.g. Circonus, Statsite, or plain statsd), you can find the configuration reference [here][consul-telemetry-config].
+
+## Visualising Telegraf Consul Metrics
+
+There a number of ways of consuming the information from Telegraf.
+
+In the [consul-guides repository][consul-guides repository], we have an example of a 3 Node Consul Cluster with a Vault instance, sending metrics to InfludDB via Telegraf, visualized using Grafana.
```

Comment:
Can we directly link to the consul-guides repo? If so, should we wait until my PR over there is merged?

Reply:
I actually think we could avoid this, as we're not explicitly tasked with maintaining it in concert with the documentation. I think it's best to keep all the content inline as it is below and not link externally. It makes it easier for the community to contribute to it all in one place, as well.

Reply:
So, should I put the Vagrant configuration in the Consul repo here and remove it from the guides, or just remove the reference completely?

Reply:
As discussed later, we removed the direct reference, we can follow up when the change is merged into https://github.com/hashicorp/consul-guides/pull/17